cedar-policy/src/api.rs

```diff
@@ -1156,21 +1386,41 @@ impl EntityUid {
     }
 
     /// Returns the id portion of the Euid
+    /// ```
+    /// use cedar_policy::{Entity, EntityId, EntityTypeName, EntityUid};
+    /// use std::str::FromStr;
+    /// let json_data = serde_json::json!({ "__entity": { "type": "User", "id": "alice" } });
+    /// let euid = EntityUid::from_json(json_data).unwrap();
+    /// assert_eq!(euid.id(), &EntityId::from_str("alice").unwrap());
+    /// ```
     pub fn id(&self) -> &EntityId {
         EntityId::ref_cast(self.0.eid())
     }
 
     /// Creates `EntityUid` from `EntityTypeName` and `EntityId`
+    ///```
+    /// use cedar_policy::{Entity, EntityId, EntityTypeName, EntityUid};
+    /// use std::str::FromStr;
+    /// let eid = EntityId::from_str("alice").unwrap();
+    /// let type_name: EntityTypeName = EntityTypeName::from_str("User").unwrap();
+    /// let euid = EntityUid::from_type_name_and_id(type_name, eid);
+    /// assert_eq!(euid.type_name(), &EntityTypeName::from_str("User").unwrap());
+    /// assert_eq!(euid.id(), &EntityId::from_str("alice").unwrap());
+    ///
+    /// ```
     pub fn from_type_name_and_id(name: EntityTypeName, id: EntityId) -> Self {
         Self(ast::EntityUID::from_components(name.0, id.0))
     }
 
     /// Creates `EntityUid` from a JSON value, which should have
     /// either the implicit or explicit `__entity` form.
-    ///
-    /// Examples:
-    /// * `{ "__entity": { "type": "User", "id": "123abc" } }`
-    /// * `{ "type": "User", "id": "123abc" }`
+    /// ```
+    /// use cedar_policy::{Entity, EntityId, EntityTypeName, EntityUid};
+    /// use std::str::FromStr;
+    /// let json_data = serde_json::json!({ "__entity": { "type": "User", "id": "123abc" } });
+    /// let euid = EntityUid::from_json(json_data).unwrap();
+    /// assert_eq!(euid.type_name(), &EntityTypeName::from_str("User").unwrap());
```

Comment:
Update to also show the implicit example

Reply:
I skipped the implicit example, because it only works for `uid` and `parents`. If we have `"attrs":{ "friend":{"type": "User", "id": "bob"}},`, `friend` does not point to an entity uid.